examples/seismic/tti/operators.py

```diff
@@ -193,12 +199,22 @@ def kernel_centered_2d(model, u, v, space_order):
     costheta = cos(model.theta)
     sintheta = sin(model.theta)
 
-    Gxx = Gxx_centered_2d(u, costheta, sintheta, space_order)
-    Gzz = Gzz_centered_2d(v, costheta, sintheta, space_order)
-    return second_order_stencil(model, u, v, Gxx, Gzz)
+    m, damp, delta, epsilon = model.m, model.damp, model.delta, model.epsilon
+    epsilon = 1 + 2 * epsilon
+    delta = sqrt(1 + 2 * delta)
+    s = model.grid.stepping_dim.spacing
+
+    if kwargs.get('forward', True):
+        Gxx = Gxx_centered_2d(model, u, costheta, sintheta, space_order)
+        Gzz = Gzz_centered_2d(model, v, costheta, sintheta, space_order)
+        return second_order_stencil(m, damp, epsilon, delta, s, u, v, Gxx, Gzz)
+    else:
+        Gxx = Gxx_centered_2d(model, (epsilon*u + delta*v), costheta, sintheta, space_order)
+        Gzz = Gzz_centered_2d(model, (delta*u + v), costheta, sintheta, space_order)
+        return second_order_stencil(m, damp, epsilon, delta, s, u, v, Gxx, Gzz, forward=False)
```

Comment:
you would avoid the other `if/else` if you compute H0/Hz inside this one and pass it to second_order_stencil:
```
forward: H0 = 1+2epsilon Gxx _ ..., 
adjoint_: H0 = Gxx_...

second_order_stencil(model, H0, Hz)
```